src/test/java/de/tum/in/www1/artemis/service/GitlabCIServiceTest.java

```diff
@@ -0,0 +1,138 @@
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Optional;
+
+import org.gitlab4j.api.GitLabApiException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import de.tum.in.www1.artemis.AbstractSpringIntegrationGitlabCIGitlabSamlTest;
+import de.tum.in.www1.artemis.ContinuousIntegrationTestService;
+import de.tum.in.www1.artemis.domain.VcsRepositoryUrl;
+import de.tum.in.www1.artemis.domain.participation.ProgrammingExerciseStudentParticipation;
+import de.tum.in.www1.artemis.exception.GitLabCIException;
+
+public class GitlabCIServiceTest extends AbstractSpringIntegrationGitlabCIGitlabSamlTest {
+
+    @Value("${artemis.version-control.url}")
+    private URL gitlabServerUrl;
+
+    @Autowired
+    private ContinuousIntegrationTestService continuousIntegrationTestService;
+
+    @BeforeEach
+    public void initTestCase() throws Exception {
+        gitlabRequestMockProvider.enableMockingOfRequests();
+        continuousIntegrationTestService.setup(this, continuousIntegrationService);
+    }
+
+    @AfterEach
+    public void tearDown() throws IOException {
+        database.resetDatabase();
+        super.resetMockProvider();
+        super.resetSpyBeans();
+        continuousIntegrationTestService.tearDown();
+    }
+
+    @Test
+    @WithMockUser(roles = "INSTRUCTOR", username = "instructor1")
+    public void testHealth() throws Exception {
+        var health = continuousIntegrationService.health();
+        assertThat(health.isUp()).isTrue();
+        assertThat(health.getAdditionalInfo()).containsEntry("cf.", "Version Control Server").containsEntry("url", gitlabServerUrl);
+    }
+
+    @Test
+    @WithMockUser(roles = "INSTRUCTOR", username = "instructor1")
+    public void testGetWebHookUrl() {
+        // TODO
+        var result = continuousIntegrationService.getWebHookUrl(null, null);
+        assertThat(result).isEqualTo(Optional.empty());
+    }
+
+    @Test
+    @WithMockUser(roles = "INSTRUCTOR", username = "instructor1")
+    public void testGetBuildStatus() {
+        // TODO
+        var result = continuousIntegrationService.getBuildStatus(null);
+        assertThat(result).isNull();
+    }
+
+    @Test
+    @WithMockUser(roles = "INSTRUCTOR", username = "instructor1")
+    public void testTriggerBuildSuccess() throws GitLabApiException {
+        mockTriggerBuild(null);
+        ProgrammingExerciseStudentParticipation participation = new ProgrammingExerciseStudentParticipation();
+        participation.setRepositoryUrl("http://some.test.url/PROJECTNAME/REPONAME-exercise.git");
+        continuousIntegrationService.triggerBuild(participation);
```

Comment:
Missing assertions about the values with which the mock was called?

Reply:
I'm not quite sure what you mean and how to implement it. I have added some `verify`s. Hope it is correct 😀.

Reply:
Sorry for not being clear. That is exactly what I meant. :+1: